### PR TITLE
:memo: simplify usability of documentation

### PR DESCRIPTION
**Pull request by fabienjuif**
@bpetetot can you review it please ?
visible here: https://github.com/alakarteio/k-redux-factory/blob/fabienjuif-patch-1/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,8 @@ Factory of Redux reducers and their associated actions and selectors.
  - [Purpose](#purpose)
  - [Why ?](#why)
  - [Installation](#installation)
- - [API](#api)
+ - [Documentation](#api)
+ - [API](./TYPES.md)
 
 ## Purpose
 `k-redux-factory` creates generic reducers, actions and selectors in two lines.
```
Comment (bpetetot) [inline README.md]: You should add some example of actions and selectors. and a link to the API for more info

**Comment by bpetetot** [comment]
In the `TYPE.md` file, links doesn't works : 
```
keyValue
simpleObject
```

And I think you should add `options` description in this file.
This file should be a complete API reference file.

**Comment by fabienjuif** [comment]
@bpetetot new commits thanks to your previous review

File: TYPES.md [hunk 2]
```diff
@@ -1,7 +1,36 @@
 # Types
-
- - [keyValue](#keyValue)
- - [simpleObject](#simpleObject)
+ - [Factories](#factories)
+ - [keyValue](#keyvalue)
+ - [simpleObject](#simpleobject)
+
+# Factories
+Factories is used to create a reducer with its selectors and its actions.
```
Comment (bpetetot) [inline TYPES.md]: Factories **are** used to create a reducer with its selectors and its actions.

File: TYPES.md [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+
+There are multiple factories signatures, take you preferred between :
```
Comment (bpetetot) [inline TYPES.md]: There are multiple factories signatures, take **your favorite** between :

File: TYPES.md [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+     - if empty, the reducer will be register at the root level of the redux state
+     - you can use dot notation, like `api.raw`: your reducer will be combined into `state.api.raw.<your_reducer>`
+   - **name** (mandatory), the reducer name (for instance: `todos`)
+     - it's used to generate actions types
+     - it's used to retrieve informations from selectors
+   - **prefix** (optional) is added to actions to avoid some collisions when there are two reducers with the same name in two distincts paths
+   - **type** (optional) can be `keyValue` or `simpleObject` (default is `keyValue`)
+   - **defaultData** (optional), for `simpleObject` only, set the default data value, used by `reset` action and for initialisation (default is an empty object `{}`)
+
+You can see documentation about specific types there:
```
Comment (bpetetot) [inline TYPES.md]: You can see documentation about specific **factories** there:

File: TYPES.md [hunk 5]
```diff
@@ -19,6 +48,22 @@ import factory from 'k-redux-factory'
 export default factory({ type: 'keyValue', /* other options */ })
 ```
 
+There are multiple factories signatures, take you preferred between :
```
Comment (bpetetot) [inline TYPES.md]: There are multiple factories signatures, take **your favorite** between :

File: TYPES.md [hunk 6]
```diff
@@ -70,6 +115,23 @@ import factory from 'k-redux-factory'
 export default factory({ type: 'simpleObject', /* other options */ })
 ```
 
+There are multiple factories signatures, take you preferred between :
```
Comment (bpetetot) [inline TYPES.md]: There are multiple factories signatures, take **your favorite** between :

File: TYPES.md [hunk 7]
```diff
@@ -19,6 +48,22 @@ import factory from 'k-redux-factory'
...
+ - `keyValue(middlewares)(options)` : this is a `keyValue`  factory with middlewares
+ - `keyValue(options)` : this is a `keyValue`  factory without middlewares
+
+Parameters are :
+ - **middlewares** (optional), contain an object with `pre` and `post` fields. Both are an array of middlewares to apply before and after the `core` middleware
+ - **options** (mandatory), either a string representating the reducer `name`, either an object with these fields :
+   - **key** (optional), the field used to identify your objects (`id` for example)
```
Comment (bpetetot) [inline TYPES.md]: this is not mandatory for `keyValues` ? what is the default key name if nothing is provided ?

Comment (fabienjuif) [reply TYPES.md]: You are right, there is not default value anymore.

File: TYPES.md [hunk 8]
```diff
@@ -1,7 +1,36 @@
...
+ - `factory(options)` : this is the root factory, without middlewares
+ - `simpleObject(middlewares)(options)` : this is a `simpleObject` factory with middlewares
+ - `simpleObject(options)` : this is a `simpleObject` factory without middlewares
+ - `keyValue(middlewares)(options)` : this is a `keyValue`  factory with middlewares
+ - `keyValue(options)` : this is a `keyValue`  factory without middlewares
+
+Parameters are :
+ - **middlewares** (optional), contain an object with `pre` and `post` fields. Both are an array of middlewares to apply before and after the `core` middleware
+ - **options** (mandatory), either a string representating the reducer `name`, either an object with these fields :
+   - **key** (mandatory), the field used to identify your objects (`id` for example)
```
Comment (bpetetot) [inline TYPES.md]: here, it's mandatory only for `type: keyValue`

File: TYPES.md [hunk 9]
```diff
@@ -17,7 +17,7 @@ There are multiple factories signatures, take you favorite between:
 Parameters are :
  - **middlewares** (optional), contain an object with `pre` and `post` fields. Both are an array of middlewares to apply before and after the `core` middleware
  - **options** (mandatory), either a string representating the reducer `name`, either an object with these fields :
-   - **key** (mandatory), the field used to identify your objects (`id` for example)
+   - **key** (mandatory for `keyValue` type), the field used to identify your objects (`id` for example)
```
Comment (guillaumecrespel) [inline TYPES.md]: It's use only for keyValue, not use for simpleObject

**Pull request merged**